test_apps/smokeTests/gridTools.cpp

```diff
@@ -54,9 +54,10 @@ template<typename T> vector<T *> AllocateBlocksType(const vector<size_t> &bs, co
         nblocks *= nb;
     }
 
-    T *buf = new T[nblocks * block_size];
+    void *tmp = std::malloc(sizeof(T) * nblocks * block_size);
```

Comment:
I'm curious. Can you explain how swapping malloc for new fixes the problem?

Reply:
The old problem is a mismatch between `new T[]` and `delete`. The fix uses malloc and free, which are a perfect pair.